lib/spack/spack/installer.py

```diff
@@ -1716,6 +1716,16 @@ def install(self):
                 spack.hooks.on_install_cancel(task.request.pkg.spec)
                 raise
 
+            except binary_distribution.NoChecksumException as exc:
```

Comment:
There are three places where this exception is raise, two occur during tarball extraction when there is a mismatch with the error being:

```
Package tarball failed checksum verification.
It cannot be installed.
```

The third is a verification of the sha256 for the tarball, with the error reported as:

```
cannot verify checksum for "<tarball-path>" [expected=<sha256>]
``` 

Should there be a distinction between the two failures (e.g., perhaps at least warning for the latter)?

Reply:
From my perspective it does not matter where the error comes from as this is just a safety net in case something is wrong with the cheksum. If you would rather see warnings relating to the original error message then that's fine as well.